lib/ansible/modules/cloud/amazon/dynamodb_table.py

```diff
@@ -216,13 +242,33 @@ def create_or_update_dynamo_table(connection, module):
         if not module.check_mode:
             result['table_status'] = table.describe()['Table']['TableStatus']
 
+        if tags:
+            # only tables which are active can be tagged
+            wait_until_table_active(module, boto3_dynamodb, table, wait_for_active_timeout)
+            account_id = get_account_id(boto3_iam)
+            boto3_dynamodb.tag_resource(ResourceArn='arn:aws:dynamodb:' + region + ':' + account_id + ':table/' + table_name, Tags=ansible_dict_to_boto3_tag_list(tags))
+
     except BotoServerError:
         result['msg'] = 'Failed to create/update dynamo table due to error: ' + traceback.format_exc()
         module.fail_json(**result)
     else:
         module.exit_json(**result)
 
 
+def get_account_id(boto3_iam):
+    users = boto3_iam.list_users(MaxItems=1)
+    return users['Users'][0]['Arn'].split(':')[4]
```

Comment:
This might be a simpler way to get the AWS account ID.  I quickly tested that it works.
```
boto3_sts = boto3_conn(module, conn_type='client', resource='sts', region=region, endpoint=ec2_url, **aws_connect_kwargs)
...
account_id = boto3_sts.get_caller_identity()["Account"]
```

It also means we don't require the `iam:ListUsers` AWS permission for this ansible module to work.